src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+        visualization (`None` or List[PIL.Image.Image]):
+            Colorized predictions for visualization.
+        uncertainty (`None`, `np.ndarray`, `torch.FloatTensor`):
+            Uncertainty maps computed from the ensemble. The shape is $numimages \times 1 \times height \times width$.
+        latent (`None`, `torch.FloatTensor`):
+            Latent features corresponding to the predictions. The shape is $numimages * numensemble \times 4 \times
+            latentheight \times latentwidth$.
+    """
+
+    prediction: Union[Image.Image, np.ndarray, torch.FloatTensor]
+    visualization: Union[None, Image.Image, List[Image.Image]]
+    uncertainty: Union[None, np.ndarray, torch.FloatTensor]
+    latent: Union[None, torch.FloatTensor]
+
+
+class MarigoldDepthPipeline(DiffusionPipeline):
+    """
+    Pipeline for monocular depth estimation using the Marigold method: https://marigoldmonodepth.github.io.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
+    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)
+
+    Args:
+        unet (`UNet2DConditionModel`):
+            Conditional U-Net to denoise the depth latent, conditioned on image latent.
+        vae (`AutoencoderKL`):
+            Variational Auto-Encoder (VAE) Model to encode and decode images and predictions to and from latent
+            representations.
+        scheduler (`DDIMScheduler` or `LCMScheduler`):
+            A scheduler to be used in combination with `unet` to denoise the encoded image latents.
+        text_encoder (`CLIPTextModel`):
+            Text-encoder, for empty text embedding.
+        tokenizer (`CLIPTokenizer`):
+            CLIP tokenizer.
+    """
+
+    model_cpu_offload_seq = "text_encoder->vae.encoder->unet->vae.decoder"
+
+    def __init__(
+        self,
+        unet: UNet2DConditionModel,
+        vae: AutoencoderKL,
+        scheduler: Union[DDIMScheduler, LCMScheduler],
+        text_encoder: CLIPTextModel,
+        tokenizer: CLIPTokenizer,
+        default_denoising_steps: Optional[int] = None,
+        default_processing_resolution: Optional[int] = None,
+    ):
+        super().__init__()
+
+        self.register_modules(
+            unet=unet,
+            vae=vae,
+            scheduler=scheduler,
+            text_encoder=text_encoder,
+            tokenizer=tokenizer,
+        )
+        self.register_to_config(
+            default_denoising_steps=default_denoising_steps,
+            default_processing_resolution=default_processing_resolution,
+        )
+
+        self.vae_scale_factor = 8
+        self.latent_space_size = self.vae.config.latent_channels
+        self.latent_scaling_factor = self.vae.config.scaling_factor
+        self.default_denoising_steps = default_denoising_steps
+        self.default_processing_resolution = default_processing_resolution
```

Comment:
ok with these special attributes here if this is what you prefer, but all these variables are accessible via `self.config` and normally we will just use that.

in general, I find it more readable to use standardized variable names as our users are already familiar with what they are.

Reply:
1. `ensemble_depth`: This method is not supposed to be called by a user. Please see this discussion to help decide if I can keep using asserts to enforce protocols: https://github.com/huggingface/diffusers/pull/7847#discussion_r1601867190
2. `ensemble_size`: Done
3. No shortcutting `self.config` - will do in the next commit, thanks!

Reply:
Done